### PR TITLE
[Microsoft Sentinel Hackathon Spring 2022] Manage secure multi cloud environment with Sentinel

### PR DESCRIPTION
**Pull request by tungnt77**
Building and utilizing multi cloud environment nowadays is the best choice for most of organization and enterprise when they start their cloud journey. Therefore, in day-to-day work of a Microsoft Sentinel Consulting team, we are often challenged by clients about how to leverage Sentinel as a good-to-go solution for managing the security events in their multi cloud environment, especially AWS and Azure. So when we started this hackathon, we want to solve this problem completely but in a "cloud native" way.

When we looked up existing applications on Sentinel marketplace for connecting Sentinel to AWS environment and AWS security specific services (Security Hub, Inspector) we found that they have some limited

Common integration is using API, causing many issues when running into high load production environment.
S3 integration is new upgrade but it is still in preview and support a little of services: VPC Flow Logs, GuardDuty and CloudTrail.
All of these problems inspired us to building an end-to-end security solution to leverage Microsoft Sentinel as a core component for a secure multi cloud environment. We want to use Sentinel to collect security findings in some AWS security services by using S3 bucket for log integration that we call an AWS native approach.

### CODE REVIEW

**Comment by v-sabiraj** [comment]
Hey @tungnt77 , can you please add the sample data, for the tables mentioned i.e Macie_CL and AWS_Inspector_CL  thanks.

File: Solutions/AWS/Data Connectors/azuredeploy_AWSConnector_API_FunctionApp.json [hunk 1]
```diff
@@ -0,0 +1,163 @@
+{
```
Comment (v-sabiraj) [inline Solutions/AWS/Data Connectors/azuredeploy_AWSConnector_API_FunctionApp.json:1]: Hey @tungnt77, Please check the [template  ](https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/azuredeploy_DataConnector_API_AzureFunctionApp_template.json) and update azuredeploy file format accordingly.

File: Solutions/AWS/Data Connectors/Connector_AWSInspectorAndMacie.json [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            ]
+        },
+        {
+            "title": "",
+            "description": "**Azure Resource Manager (ARM) Template**\n\nUse this method for automated deployment of the AWS Connector.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://aka.ms/sentinel-awsinspectorandmacie-azuredeploy)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n3. Enter the **Workspace ID**, **Workspace Key**, **Qradar Authorized Service Token**, **SPN Credential**. \n>Note: If using Azure Key Vault secrets for any of the values above, use the`@Microsoft.KeyVault(SecretUri={Security Identifier})`schema in place of the string values. Refer to [Key Vault references documentation](https://docs.microsoft.com/azure/app-service/app-service-key-vault-references) for further details. \n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**. \n5. Click **Purchase** to deploy."
+        }
+        
+    ],
+    "metadata": {
+        "id": "Unique Identifier (GUID) used to identify dependencies and content from solutions or community.",
```
Comment (v-sabiraj) [inline Solutions/AWS/Data Connectors/Connector_AWSInspectorAndMacie.json:112]: Hey @tungnt77, please update the id with proper GUID value, thanks.

File: Solutions/AWS/Data Connectors/Connector_AWSInspectorAndMacie.json [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        "version": "1.0.0",
+        "kind": "dataConnector",
+        "source": {
+            "kind": "source type of the content. Value must be one of these : localWorkspace | community | solution | sourceRepository",
```
Comment (v-sabiraj) [inline Solutions/AWS/Data Connectors/Connector_AWSInspectorAndMacie.json:116]: @tungnt77, please add the source details (value for kind and name), thanks.

**Review by v-sabiraj — CHANGES_REQUESTED** [review]

**Comment by v-sabiraj** [comment]
Hi @tungnt77, please add the zip file for data connector, and also check the folder structure for data connector, you can refer [sample connector](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Box/Data%20Connectors), just for the refence, thanks.

**Comment by v-sabiraj** [comment]
Hey @tungnt77, please resolve the requested changes, thanks.

**Comment by v-spadarthi** [comment]
@tungnt77, please resolve the requested changes and validation checks as well, thanks.

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Please fix the validation issues before this can be approved.

The Solution uses custom tables like AWS_Inspector_CL and Macie_CL. We probably need to define this in the custom table. Please see an example here: https://github.com/Azure/Azure-Sentinel/blob/master/.script/tests/KqlvalidationsTests/CustomTables/AWSGuardDuty.json

Please also include the **entityMappings** and **kind** fields for the analytic templates

**Comment by vmanojreddy** [comment]
@tungnt77 Please address the above comments. thanks!!!

**Comment by v-sabiraj** [comment]
@tungnt77 , please update the requested changes. Thanks.

**Comment by v-spadarthi** [comment]
@tungnt77, please address the requested changes and validation checks as well, thanks.

**Comment by v-spadarthi** [comment]
@tungnt77, please address the requested changes and validation checks as well, thanks.

**Comment by pemontto** [comment]
@v-spadarthi @v-sabiraj @v-marimanda 
Seems like shouting into a void, would Microsoft consider just taking on the remaining tasks to get some much needed additional functionality into Sentinel?

**Comment by v-marimanda** [comment]
Hi @tungnt77,
We wanted to check on the status of PR [4944](https://github.com/Azure/Azure-Sentinel/pull/4944). PR is pending from long time. Let us know if any assistance is required for this PR. Per our standard operating procedures if no response is received in the next 7 business days we will close this PR. Thank you for your cooperation.

**Comment by NikTripathi** [comment]
@tungnt77 Can you please provide some update. @v-marimanda Please do the needful.

**Comment by v-marimanda** [comment]
Hi @tungnt77,
Since we have not received response from last 10 days, we are closing your PR #4944 per our standard operating procedures. If you still need support for this PR you can re-open the PR at any time.
If you do re-open, we simply request that you ensure the PR has response to the last request. Thank you for your cooperation.

**Comment by v-mchatla** [comment]
Hi @v-sabiraj,
Can you please provide update on this PR
Thanks

**Comment by v-mchatla** [comment]
Hi @v-sabiraj 
Can you please provide update on this.
Thanks

**Comment by v-sabiraj** [comment]
Hi, need to work on this, let you know once done.

**Comment by v-mchatla** [comment]
Hi @v-sabiraj,
Please provide an update on the progress once you started working on it.
Thanks

**Comment by v-sabiraj** [comment]
Hi, will start working on this soon.

**Comment by v-spadarthi** [comment]
Hi @v-sabiraj,
Please provide an update on the progress once you started working on it.
Thanks

**Comment by v-sabiraj** [comment]
Closing this for now, will reopen once we start working.

**Pull request closed without merging**